### PR TITLE
Add monitoring metrics for database queries

### PR DESCRIPTION
**Pull request by dkatzz**
### Description

Add monitoring metrics for database queries, including query counts and mean, max and total latency

## Release notes

Add monitoring metrics for database queries that can be consumed by AWS grafana

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >

### Issue(s) this completes

#4812

### CODE REVIEW

File: server/app/controllers/monitoring/MetricsController.java [hunk 1]
```diff
@@ -40,6 +74,17 @@ public Result getMetrics() {
     var writer = new StringWriter();
 
     try {
+      ServerMetrics serverMetrics = database.getMetaInfoManager().collectMetrics();
+      serverMetrics
```
Comment (bion) [inline server/app/controllers/monitoring/MetricsController.java:78]: **nitpick:** single line?
```suggestion
      database.getMetaInfoManager().collectMetrics()
```

File: server/app/controllers/monitoring/MetricsController.java [hunk 2]
```diff
@@ -40,6 +74,17 @@ public Result getMetrics() {
...
+          .getQueryMetrics()
+          .forEach(
+              metric -> {
+                String name = metric.getName().substring(4);
```
Comment (bion) [inline server/app/controllers/monitoring/MetricsController.java:82]: **suggestion (blocking):** extract 4 to a constant and add a comment for why it's 4 and not some other value

File: server/app/views/FileUploadViewStrategy.java [hunk 3]
```diff
@@ -104,7 +104,7 @@ protected abstract ImmutableList<InputTag> fileUploadFields(
   /**
    * Method to render the UI for uploading a file.
    *
-   * @param params the information needed to render a file upload view
+   * @param params the information needed to rener a file upload view
```
Comment (bion) [inline server/app/views/FileUploadViewStrategy.java]: **nitpick:** typo

Comment (dkatzz) [reply server/app/views/FileUploadViewStrategy.java]: Should be from an old commit - I accidentally modified this file :)

File: server/app/controllers/monitoring/MetricsController.java (hunk 1 above)
Comment (dkatzz) [reply server/app/controllers/monitoring/MetricsController.java]: Done

File: server/app/controllers/monitoring/MetricsController.java (hunk 2 above)
Comment (dkatzz) [reply server/app/controllers/monitoring/MetricsController.java]: Done

**Comment by dkatzz** [comment]
> Is there much in the way of options for testing this controller? Understandable if not but it'd be nice to exercise the code if possible

Added a test for the MetricsController - let me know what you think!

**Review by bion — APPROVED** [review]

File: server/test/controllers/monitoring/MetricsControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
+package controllers.monitoring;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.test.Helpers.*;
```
[the remark this answers is not in the document]
Comment (dkatzz) [reply server/test/controllers/monitoring/MetricsControllerTest.java]: Updated

File: server/app/controllers/monitoring/MetricsController.java [hunk 5]
```diff
@@ -21,11 +24,45 @@ public final class MetricsController extends CiviFormController {
 
   private final boolean metricsEnabled;
   private final CollectorRegistry collectorRegistry;
+  private final Database database;
+
+  private static final Counter queryMetricCount =
```
Comment (bion) [inline server/app/controllers/monitoring/MetricsController.java:29]: **nitpick:** SCREAMING_SNAKE_CASE for constants

Comment (dkatzz) [reply server/app/controllers/monitoring/MetricsController.java]: Updated! As well as made the changes we discussed.

**Pull request merged**